src/core/qfieldxmlhttprequest.h

```diff
@@ -0,0 +1,314 @@
+*   it under the terms of the GNU General Public License as published by  *
+*   the Free Software Foundation; either version 2 of the License, or     *
+*   (at your option) any later version.                                   *
+*                                                                         *
+***************************************************************************/
+
+#ifndef QFIELDXMLHTTPREQUEST_H
+#define QFIELDXMLHTTPREQUEST_H
+
+#include <QJSValue>
+#include <QList>
+#include <QMap>
+#include <QMetaObject>
+#include <QMetaType>
+#include <QNetworkReply>
+#include <QNetworkRequest>
+#include <QObject>
+#include <QPointer>
+#include <QUrl>
+#include <QVariant>
+
+class NetworkReply;
+class QHttpMultiPart;
+class QNetworkReply;
+
+/**
+ * \brief QML-friendly XMLHttpRequest using QField's NetworkManager/NetworkReply.
+ *
+ * This class provides a small subset of the browser XMLHttpRequest (XHR) workflow,
+ * but implemented on top of QField's networking stack:
+ *
+ *   - open(method, url)
+ *   - setRequestHeader(name, value)
+ *   - send(body)
+ *   - abort()
+ *
+ * Observability:
+ *   - readyState/status/statusText/responseText/response/responseType/responseUrl
+ *   - callback properties: onreadystatechange, ondownloadprogress, onuploadprogress,
+ *     onredirected, ontimeout, onaborted, onerror, onload, onloadend
+ *
+ * Payload handling:
+ *   - QByteArray: sent as-is
+ *   - QString: UTF-8 text payload (Content-Type inferred if not set)
+ *   - QVariantMap/QVariantList: encoded as JSON payload (Content-Type inferred if not set)
+ *   - Multipart upload:
+ *       - enabled if body contains explicit "file://..." URLs OR Content-Type starts with "multipart/form-data"
+ *
+ * Security/safety:
+ *   - local file uploads are intentionally restricted to:
+ *       - current project directory
+ *       - QFieldCloud local directory
+ *
+ * Notes:
+ *   - The implementation aims to be minimal and predictable for QML usage, not a full browser XHR
+ */
+class QFieldXmlHttpRequest : public QObject
+{
+    Q_OBJECT
+
+    Q_PROPERTY( ReadyState readyState READ readyState NOTIFY readyStateChanged )
+    Q_PROPERTY( int status READ status NOTIFY responseChanged )
+    Q_PROPERTY( QString statusText READ statusText NOTIFY responseChanged )
+
+    // Response body decoded as UTF-8 text
+    Q_PROPERTY( QString responseText READ responseText NOTIFY responseChanged )
+    // Parsed response if JSON, otherwise equals responseText
+    Q_PROPERTY( QVariant response READ response NOTIFY responseChanged )
+    Q_PROPERTY( QString responseType READ responseType NOTIFY responseChanged )
+    Q_PROPERTY( QUrl responseUrl READ responseUrl NOTIFY responseChanged )
+
+    /**
+   * \brief Timeout in milliseconds. 0 disables
+   *    * Timeout is implemented as an external timer that aborts the request if it is still running
+   */
+    Q_PROPERTY( int timeout READ timeout WRITE setTimeout NOTIFY timeoutChanged )
```

Comment:
Either document all Q_PROPERTY, or document all the getter / setter functions.

Also, we don't use \brief in the property / function documentation. The timeout documentation here feels broken :)